docs/scos/dev/upgradability-services/new-documentation/method-of-extended-class-overridden-on-project-level.md

```diff
@@ -0,0 +1,135 @@
+    {
+        ...
+    }
+}
+```
+
+#### Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------
+************************************************************************************************************************
+Evaluator\Business\Check\IsMethodOverridden\EntityManagerCheck
+Introduce a new custom method without usage of existing one. Override usage of the current method in all usage of public API.
+************************************************************************************************************************
+------------------------------------------------------------------------------------
+Pyz\Zed\EvaluatorSpryker\Persistence\EvaluatorSprykerCategoryImageEntityManager
+{"name":"saveCategoryImageSet","class":"Pyz\\Zed\\EvaluatorSpryker\\Persistence\\EvaluatorSprykerCategoryImageEntityManager"}
+{"parentClass":"Spryker\\Zed\\CategoryImage\\Persistence\\CategoryImageEntityManager"}
+************************************************************************************************************************
+```
+
+{% info_block warningBox "Dependency Provider exception" %}
+
+If you override a method and initialize a plugin, it does not break backward compatibility. For example, in `StorageRouterDependencyProvider`, `SprykerShopStorageRouterDependencyProvider` is overridden with a plugin introduced:
+
+<details>
+<summary markdown='span'>Example of an overridden method with a plugin</summary>
+
+```php
+<?php
+
+namespace Pyz\Yves\StorageRouter;
+
+use SprykerShop\Yves\CatalogPage\Plugin\StorageRouter\CatalogPageResourceCreatorPlugin;
+use SprykerShop\Yves\CmsPage\Plugin\StorageRouter\PageResourceCreatorPlugin;
+use SprykerShop\Yves\ProductDetailPage\Plugin\StorageRouter\ProductDetailPageResourceCreatorPlugin;
+use SprykerShop\Yves\ProductSetDetailPage\Plugin\StorageRouter\ProductSetDetailPageResourceCreatorPlugin;
+use SprykerShop\Yves\RedirectPage\Plugin\StorageRouter\RedirectResourceCreatorPlugin;
+use SprykerShop\Yves\StorageRouter\StorageRouterDependencyProvider as SprykerShopStorageRouterDependencyProvider;
+
+class StorageRouterDependencyProvider extends SprykerShopStorageRouterDependencyProvider
+{
+    /**
+     * @return \SprykerShop\Yves\StorageRouterExtension\Dependency\Plugin\ResourceCreatorPluginInterface[]
+     */
+    protected function getResourceCreatorPlugins(): array
+    {
+        return [
+            new PageResourceCreatorPlugin(),
+            new CatalogPageResourceCreatorPlugin(),
+            new ProductDetailPageResourceCreatorPlugin(),
+            new ProductSetDetailPageResourceCreatorPlugin(),
+            new RedirectResourceCreatorPlugin(),
+        ];
+    }
+}
+```
+
+</details>
+
+{% endinfo_block %}
+
+#### Steps to solve potential problem in functionality
+
+1. Introduce a new custom method.
```

Comment:
1. Investigate if it is possible to extend functionality with configuration or plugins (link to "Configuration", "Plug and Play strategy" - https://docs.spryker.com/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html#plug-and-play)

Reply:
Chek in all the documentation